lms/djangoapps/teams/api.py

```diff
@@ -264,16 +265,34 @@ def get_team_count_query_set(topic_id_set, course_id, organization_protection_st
         filter_query.update(
             {'organization_protected': organization_protection_status == OrganizationProtectionStatus.protected}
         )
-    return CourseTeam.objects.filter(**filter_query)
+    return filter_query
 
 
-def add_team_count(topics, course_id, organization_protection_status):
+def get_teams_with_visibility(user, topic_id_set, course_id, organization_protection_status):
+    """ Get teams taking into account user visibility privelages """
+    # Filter by topics, course, and protection status
+    filter_query = _get_team_filter_query(topic_id_set, course_id, organization_protection_status)
+
+    # Staff gets unfiltered list of teams
+    course_module = modulestore().get_course(course_id)
+    if has_access(user, 'staff', course_module.id):
```

Comment:
nit: modulestore() is expensive, could we do this only when we have to, after checking that the user isn't staff?

Reply:
Moved it w/in this function but it does bring up the larger point of performance: should we look into caching this, or carrying the data into the Django model to streamline querying (see [other discussion](https://github.com/edx/edx-platform/pull/24216#discussion_r441680062))?

I hesitate to act on this now w/out any performance data but it's still an important topic.

Reply:
That's a much, much bigger discussion. I wouldn't be opposed to getting into it at some point, though.